### ISSUE TITLE
[discussion] Consider dropping support for Python 2

### ISSUE THREAD

**Issue by gatesn** [palantir/python-language-server#419]

**Comment by quarthex** [comment]
On the one hand, dropping this support may simplify this tool, making it easier to maintain.
I totally agree with that.
On the other hand, this would make it unavailable for projects that rely on this Python version.
You may reply that such projects should upgrade their code base to a more recent version of Python, using tools like 2to3. This is not always that easy.
Sometimes, the project is far too complex (let's be honest: with a lot of crap within) to be ported to Python 3 without taking a costly risk.
Sometimes, the project rely on Python 2 binary extensions and the sources are lost.
Sometimes, the team that maintain the library can do the jump, but the library is used by another team that won't do the same.

**Comment by kfkonrad** [comment]
I don't think this is question of whether we should, but rather when we will drop support. The support for Python 2 will end in 1 year 2 months and 23 days ([see this countdown](https://pythonclock.org)). Maybe it makes sense to maintain support for Python 2 unit then and see how well the migrations proceed until then. Maybe (hopefully) demand for Python 2 will drop.

The way I see it everyone knew that eventually Python 2 support must end and they had/still have enough time to migrate projects. At some point in time work required to maintain Python 2 will outweigh the demand for Python 2. Then this project will have a good reason to drop support for a near antique Python version.
